### PR TITLE
Identity link with SSO nodejs

### PR DESCRIPTION
**Pull request by NikitaJain-MSFT**

### CODE REVIEW

File: samples/app-identity-linking-with-sso/nodejs/README.md [hunk 1]
```diff
@@ -39,12 +45,14 @@ Messaging Extention mapping
...
   ![ME search](Images/MESearch.png)
 
-  ![ME search](Images/MESearchLinkCard.jpg)
+  ![ME search](Images/MESearchLinkCard.png)
    
 - ME Link unfurling
 
   ![ME Link unfurling](Images/MELinkUnfurlingCard.png)
 
+- On Click of `Disconnect from facebook` or `Disconnect from google`. User's facebook/google data will be removed from the data base.
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/README.md:54]: Database
Also, are we using any database or just saving it in memory.
So we can add a extensibility note like, you can use any memory like DB or in memory or in-file.

File: samples/app-identity-linking-with-sso/nodejs/index.js [hunk 2]
```diff
@@ -293,7 +293,7 @@ server.post('/GetUserDetails', async (req, res) => {
 });
 
 // Facebook Oauth token axchange
-server.post('/getFbAccessToken', function (req, res) {
+server.post('/getFbDetailsOauth', function (req, res) {
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/index.js:296]: Are we getting Oauth or just FB user details?

File: samples/app-identity-linking-with-sso/nodejs/index.js [hunk 3]
```diff
@@ -343,7 +343,7 @@ server.post('/getFbAccessToken', function (req, res) {
   });
 });
 
-server.post('/getGoogleAccessToken', function (req, res) {
+server.post('/getGoogleDetailsOauth', function (req, res) {
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/index.js:346]: Are we getting Oauth or just Google account user details?

File: samples/app-identity-linking-with-sso/nodejs/index.js [hunk 4]
```diff
@@ -390,7 +390,13 @@ server.post('/getGoogleAccessToken', function (req, res) {
 });
 
 server.post('/getGoogleDetails', function (req,res){
-  var token = req.body.token;
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/index.js:393]: What token is this?
AccessToken or ?

File: samples/app-identity-linking-with-sso/nodejs/index.js [hunk 5]
```diff
@@ -411,7 +417,13 @@ server.post('/getGoogleDetails', function (req,res){
 })
 
 server.post('/getFbDetails',function (req,res){
-  var token = req.body.token;
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/index.js:414]: What token is this?
AccessToken or ?

File: samples/app-identity-linking-with-sso/nodejs/views/tab.ejs [hunk 6]
```diff
@@ -127,12 +127,12 @@
 
                             if (userDetails.is_google_signed_in) {
                                 is_google_signed_in = userDetails.is_google_signed_in
-                                getGoogleDetails(userDetails.google_token);
+                                getGoogleDetails();
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/views/tab.ejs:130]: We are not getting google details :)
We are getting the user details for a google account =>getGoogleAccountDetails();

File: samples/app-identity-linking-with-sso/nodejs/views/tab.ejs [hunk 7]
```diff
@@ -198,34 +198,35 @@
...
+                        }
+                    })
+                    .then((responseJson) => {
+                        if (responseJson.error) {
+                            reject(responseJson.error);
+                        } 
+                        else {
+                            const userDetails = responseJson;
+                            $("#sso").hide();
+                            $("#userImgSso").attr("src", userDetails.image);
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/views/tab.ejs:215]: Naming conversions are not correct for element id's like 
$("**#sso**").hide();
$("**#work**").append(userDetails.details.jobTitle);

File: samples/app-identity-linking-with-sso/nodejs/views/tab.ejs [hunk 8]
```diff
@@ -323,7 +324,7 @@
         function getServerSideTokenFb(clientSideToken) {
             return new Promise((resolve, reject) => {
                 microsoftTeams.getContext((context) => {
-                    fetch('/getFbAccessToken', {
+                    fetch('/getFbDetailsOauth', {
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/views/tab.ejs:327]: getFbUserDetails

File: samples/app-identity-linking-with-sso/nodejs/views/tab.ejs [hunk 9]
```diff
@@ -394,7 +395,7 @@
         function getServerSideTokenGoogle(clientSideToken) {
             return new Promise((resolve, reject) => {
                 microsoftTeams.getContext((context) => {
-                    fetch('/getGoogleAccessToken', {
+                    fetch('/getGoogleDetailsOauth', {
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/views/tab.ejs:398]: getGoogleAccountDetails

File: samples/app-identity-linking-with-sso/nodejs/views/tab.ejs [hunk 10]
```diff
@@ -432,52 +433,56 @@
...
-                        $("#userImgGoogle").attr("src", userDetails.photos[0].url);
-                        $("#gname").append(userDetails.names[0].displayName);
-                        $("#gemail").append(userDetails.emailAddresses[0].value);
-                        $("#google-disConnect").show();
-                        $("#googlelogin").hide();
+        function getGoogleDetails() {
+            microsoftTeams.getContext((context) => {
+                $.ajax({
+                    type: 'POST',
+                    url: '/getGoogleDetails',
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/views/tab.ejs:440]: getGoogleAccountDetails or getGoogleAccountUserDetails

File: samples/app-identity-linking-with-sso/nodejs/views/tab.ejs [hunk 11]
```diff
@@ -432,52 +433,56 @@
...
-                },
-                error: function (xhr, textStatus, errorThrown) {
-                    console.log("textStatus: " + textStatus + ", errorThrown:" + errorThrown);
-                    fbAuth();
-                }
+        function getFbDetails() {
+            microsoftTeams.getContext((context) => {
+                $.ajax({
+                    type: 'POST',
+                    url: '/getFbDetails',
```
Comment (v-chetsh) [inline samples/app-identity-linking-with-sso/nodejs/views/tab.ejs:470]: getFbAccountDetails or getFbUserDetails

**Review by v-chetsh — COMMENTED** [review]
Reviewed and provided my comments.

File: samples/app-identity-linking-with-sso/nodejs/README.md [hunk 12]
```diff
@@ -23,10 +23,16 @@ Bot Mapping
 
 ![Bot](Images/BotLinkingOptionCard.png)
 
+- Click on `Connect to facebook` or `Connect to google` button you will get a sign in card. Click on sign in button and provide your facebook or google credentials.
```
[the remark this answers is not in the document]
Comment (NikitaJain-MSFT) [reply samples/app-identity-linking-with-sso/nodejs/README.md]: added section as about the sample

File: samples/app-identity-linking-with-sso/nodejs/index.js (hunk 3 above)
Comment (NikitaJain-MSFT) [reply samples/app-identity-linking-with-sso/nodejs/index.js]: we are getting google account details following Outh flow 
updated route as per suggestion

**Review by v-chetsh — APPROVED** [review]
Changes looks fine, approving it.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**